src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,172 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    int size;
+    Node<T> head;
+    Node<T> tail;
+
+    public MyLinkedList() {
+        this.head = null;
+        this.tail = null;
+    }
+
     @Override
     public void add(T value) {
-
+        if (head == null) {
+            head = new Node(value, null, null);
+            tail = head;
+            size = 1;
+        } else {
+            Node newTail = new Node(value, tail, null);
+            tail.next = newTail;
+            tail = newTail;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == size) {
+            linkTail(value);
+        } else {
+            link(value, node(index));
+        }
+    }
 
+    private void linkTail(T value) {
+        Node<T> prev = tail;
+        Node<T> newNode = new Node<T>(value, prev, null);
+        tail = newNode;
+        if (prev == null) {
+            head = newNode;
+        } else {
+            prev.next = newNode;
+        }
+        size++;
+    }
+
+    private void link(T value, Node<T> x) {
+        Node<T> prev = x.prev;
+        Node<T> newNode = new Node<T>(value, prev, x);
+        x.prev = newNode;
+        if (prev == null) {
+            head = newNode;
+        } else {
+            prev.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).item;
     }
 
     @Override
     public void set(T value, int index) {
+        checkIndex(index);
+        Node ptr = node(index);
+        ptr.item = value;
+    }
+
+    private void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
 
+    private Node<T> node(int index) {
```

Comment:
@Sasha192 please add more informative method name. All method names should starts from a verb